skyportal/handlers/api/analysis.py

```diff
@@ -1034,6 +1056,22 @@ async def post(self, analysis_resource_type, resource_id, analysis_service_id):
                     f'Invalid analysis_parameters: {analysis_parameters}.', status=400
                 )
 
+            if analysis_service.is_summary:
+                user_id = self.associated_user_object.id
+                user = session.scalars(
+                    User.select(session.user_or_token, mode="update").where(
+                        User.id == user_id
+                    )
+                ).first()
+                if user is None:
+                    return self.error('Cannot find user.', status=400)
+                if "summary" in user.preferences:
+                    if "OpenAI" in user.preferences["summary"]:
+                        if user.preferences["summary"]["OpenAI"]["active"]:
+                            analysis_parameters["openai_api_key"] = user.preferences[
```

Comment:
same comment as above, and it will avoid giving you errors with inaccessible keys.